src/dodal/devices/shutter.py

```diff
@@ -1,21 +1,41 @@
+import asyncio
 from enum import Enum
 
-from ophyd_async.core import StandardReadable
-from ophyd_async.epics.signal import epics_signal_rw
+from bluesky.protocols import Movable
+from ophyd_async.core import AsyncStatus, StandardReadable, wait_for_value
+from ophyd_async.epics.signal import epics_signal_r, epics_signal_w
 
 
-class OpenState(str, Enum):
+class OpenStateSet(str, Enum):
+    CLOSED = "close"
+    OPEN = "open"
+
+
+class OpenStateReadback(str, Enum):
     CLOSED = "closed"
     OPEN = "open"
     OPENING = "opening"
     CLOSING = "closing"
     FAULT = "fault"
 
 
-class Shutter(StandardReadable):
+class Shutter(StandardReadable, Movable):
     def __init__(self, prefix: str, name: str):
         with self.add_children_as_readables():
-            self.position = epics_signal_rw(
-                write_pv=prefix + "CTRL2", read_pv=prefix + "STA", datatype=OpenState
+            self.position_readback = epics_signal_r(
+                read_pv=prefix + "STA",
+                datatype=OpenStateReadback,
+            )
+            self.position_set = epics_signal_w(
+                write_pv=prefix + "CTRL2",
+                datatype=OpenStateSet,
```

Comment:
I don't think self.position_set should be a readback value. It should be outside of the add_children_as_readables() context

Reply:
yep